packages/e2e/artillery/artillery.ts

```diff
@@ -0,0 +1,16 @@
+export interface Context<T> {
+  _uid: string;
+  vars: T;
+}
+
+export type AfterResponse<T> = (
+  params: unknown,
```

Comment:
Have a look at how the `@types/artillery` package for inspiration:
https://www.npmjs.com/package/@types/artillery

Reply:
Why don't we use @types/artillery directly?

Reply:
It is useless (it actually defines nothing about the callbacks) and inefficient (the only _half useful_ defined type is `ScenarioContext`, but it do not includes `_uid` and `vars` is of an hard coded type).

Please take a look to [its source code](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/artillery/index.d.ts) and you'll agree this file defines much more useful and correct stuff.

Reply:
It's also [outdated](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9e20f594027285c597a96af85f913c61ba915b/types/artillery/index.d.ts#L1) (we're using v2), but my comment was more related to seeing how the request and response are typed using the Node.js built-in libs, which could be an improvement over the `unknown` types. Also, the `params` could be typed with a generic, i.e `AfterResponse<T, P>`

Reply:
Yep, I agree; the point is that (as far I was able to find on internet) there is no native built-in types for artillery callbacks.

About changing `unknown` into something specific; I think that spending time now to do this, is not a good time investment: we risk to spend time to discover something we will never use; what we have now let us to intercept calls and to control the flow.

In the case (in my opinion unlikely) in which we need to write an artillery tests which needs something else: we will need to study the _new feature_ and that will be the right occasion to enrich the types I wrote till now.

Reply:
I think that's reasonable as long as we list all fields, so devs can see that they exist and can then improve the types if needed

Reply:
Added all the fields with a description (where I was able to find... unknown otherwise)